include/dlaf/factorization/qr/t_factor_mc.h

```diff
@@ -0,0 +1,179 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2020, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <hpx/future.hpp>
+#include <hpx/include/util.hpp>
+
+#include <blas.hh>
+
+#include "dlaf/factorization/qr/api.h"
+
+#include "dlaf/common/data.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/pipeline.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/functions_sync.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace factorization {
+namespace internal {
+
+template <class T>
+struct QR_Tfactor<Backend::MC, Device::CPU, T> {
+  static void call(const SizeType k, Matrix<const T, Device::CPU>& v, const GlobalTileIndex v_start,
+                   common::internal::vector<hpx::shared_future<T>> taus, Matrix<T, Device::CPU>& t,
+                   common::Pipeline<comm::CommunicatorGrid>& serial_comm);
+};
+
+template <class T>
+void QR_Tfactor<Backend::MC, Device::CPU, T>::call(
+    const SizeType k, Matrix<const T, Device::CPU>& v, const GlobalTileIndex v_start,
+    common::internal::vector<hpx::shared_future<T>> taus, Matrix<T, Device::CPU>& t,
+    common::Pipeline<comm::CommunicatorGrid>& serial_comm) {
+  using hpx::util::unwrapping;
+  using common::make_data;
+  using namespace comm::sync;
+
+  const auto& dist = v.distribution();
+  const comm::Index2D rank = dist.rankIndex();
+  const comm::Index2D rank_v0 = dist.rankGlobalTile(v_start);
+
+  const auto panel_width = v.tileSize(v_start).cols();
+
+  DLAF_ASSERT(k <= panel_width, k, panel_width);
+
+  DLAF_ASSERT(t.nrTiles() == GlobalTileSize(1, 1), t);
+  // TODO assertion about t.blocksize() == v.blocksize()?
+  // TODO assertion about t.size() is compatible with k/panel_width?
+  // DLAF_ASSERT(t.size() == GlobalElementSize(k, k), t.size(), k);
+
+  // TODO assumption: no empty grid
+
+  if (rank.col() != rank_v0.col())
+    return;
+
+  const LocalTileIndex v_start_loc{
+      dist.template nextLocalTileFromGlobalTile<Coord::Row>(v_start.row()),
+      dist.template nextLocalTileFromGlobalTile<Coord::Col>(v_start.col()),
+  };
+  const LocalTileIndex v_end_loc{dist.localNrTiles().rows(), v_start_loc.col() + 1};
+
+  // TODO it would be better to embed this reset inside a bigger task
+  matrix::util::set(t, [](auto&&) { return 0; });
```

Comment:
I can absolutely use `laset` and I'll do, but the main point of my comment was: is it ok to have a task that just resets or do we want to try embed this reset in the first computing task in some way? @rasolca

Reply:
as discussed with @rasolca offline, `laset` of a tile is enough work for a task. we leave it as it is now.